services/storage/src/simcore_service_storage/handlers_files.py

```diff
@@ -105,13 +144,47 @@ async def upload_file(request: web.Request):
     )
 
     dsm = get_dsm_provider(request.app).get(path_params.location_id)
-    link: AnyUrl = await dsm.create_file_upload_link(
+    links: UploadLinks = await dsm.create_file_upload_links(
         user_id=query_params.user_id,
         file_id=path_params.file_id,
         link_type=query_params.link_type,
+        file_size_bytes=query_params.file_size or ByteSize(0),
     )
+    if query_params.file_size is None:
+        # return v1 response
+        assert len(links.urls) == 1  # nosec
+        return {"link": jsonable_encoder(links.urls[0], by_alias=True)}
 
-    return {"link": jsonable_encoder(link, by_alias=True)}
+    # v2 response
+    abort_url = request.url.join(
+        request.app.router["abort_upload_file"]
+        .url_for(
+            location_id=f"{path_params.location_id}",
+            file_id=urllib.parse.quote(path_params.file_id, safe=""),
+        )
+        .with_query(user_id=query_params.user_id)
+    )
+    complete_url = request.url.join(
+        request.app.router["complete_upload_file"]
+        .url_for(
+            location_id=f"{path_params.location_id}",
+            file_id=urllib.parse.quote(path_params.file_id, safe=""),
+        )
+        .with_query(user_id=query_params.user_id)
+    )
+    response = FileUploadSchema(
+        chunk_size=links.chunk_size,
+        urls=links.urls,
+        links=FileUploadLinks(
+            abort_upload=parse_obj_as(AnyUrl, f"{abort_url}"),
+            complete_upload=parse_obj_as(
+                AnyUrl,
+                f"{complete_url}",
+            ),
+        ),
+    )
+
+    return jsonable_encoder(response, by_alias=True)
```

Comment:
THOUGHTS:
we have many json-engines  and encodes out of sync, namely
- the one used in ``jsonable_encoder``
- the one in ``aiohttp.json_response``
- the one in pydantic classes (used in ``model.json()``)

sometimes it will be built-in ``json``, sometimes ``ujson`` sometimes ``orjson`` ... :-)

I wonder how we should proceed here. IMO the important thing is to have one with encoders for all serialized types.